src/components/Personalization/index.js

```diff
@@ -60,6 +77,11 @@ const createPersonalization = ({ config, logger }) => {
           componentRegistry.getCommand(EVENT_COMMAND)
         );
       },
+      onBeforeDataCollection(payload) {
+        const sessionId = getOrCreateSessionId(cookie);
```

Comment:
Because `getOrCreateSessionId` uses cookie values and writes to cookies, this probably needs to happen _after_ the user opts in. See https://github.com/adobe/alloy/blob/master/src/components/Identity/index.js#L51 for an example of how this is done.

Reply:
Thanks, I have adjusted.